Packs/XMatters/Integrations/xMatters/xMatters.py

```diff
@@ -0,0 +1,801 @@
+    reduced_out: Dict[str, List[Any]]
+    if len(out) == 0:
+        reduced_out = {"xMatters.GetEvent.Event": []}
+        readable_output = "Could not find Events with given criteria in xMatters"
+    else:
+        reduced_out = {"xMatters.GetEvents.Events": [event_reduce(event) for event in out]}
+        readable_output = f'Retrieved Events from xMatters: {reduced_out}'
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs=reduced_out,
+        outputs_prefix='xMatters.GetEvents',
+        outputs_key_field='event_id'
+    )
+
+
+def xm_get_event_command(client: Client, event_id: str) -> CommandResults:
+    """
+    This function is run when the xm-get-event command is run.
+
+    :type client: ``Client``
+    :param Client: xMatters client to use
+
+    :type event_id: ``str``
+    :param event_id: Event ID to search for in xMatters
+
+    :return: Output of xm-get-event command
+
+    :rtype: ``CommandResults``
+    """
+    out = client.search_alert(event_id=event_id)
+
+    reduced_out: Dict[str, Any]
+    if out.get('code') == 404:
+        reduced_out = {"xMatters.GetEvent.Event": {}}
+        readable_output = f'Could not find Event "{event_id}" from xMatters'
+    else:
+        reduced = event_reduce(out)
+        reduced_out = {"xMatters.GetEvent.Event": reduced}
+        readable_output = f'Retrieved Event "{event_id}" from xMatters:\nEventID: {reduced.get("EventID")}\n' \
+                          f'Created: {reduced.get("Created")}\nTerminated: {reduced.get("Terminated")}\n' \
+                          f'Name: {reduced.get("Name")}\nStatus: {reduced.get("Status")}'
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs=reduced_out,
+        outputs_prefix='xMatters.GetEvent',
+        outputs_key_field='event_id'
+    )
+
+
+def test_module(from_xm: Client, to_xm: Client, user: str) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: xMatters client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    # INTEGRATION DEVELOPER TIP
+    # Client class should raise the exceptions, but if the test fails
+    # the exception text is printed to the Cortex XSOAR UI.
+    # If you have some specific errors you want to capture (i.e. auth failure)
+    # you should catch the exception here and return a string with a more
+    # readable output (for example return 'Authentication Error, API Key
+    # invalid').
+    # Cortex XSOAR will print everything you return different than 'ok' as
+    # an error
+    try:
+        to_xm.xm_trigger_workflow(
+            recipients='nobody',
+            subject='Test - please ignore',
+            body='Test - please ignore'
+        )
```

Comment:
Every time someone presses the "Test" button for the integration a workflow is triggered in XMatters? Would this create a lot of spam? If only a `GET` call could be done that would be best.

Reply:
There is probably no user named `nobody` so the event wouldn't notify anyone. Each test would generate an event visible in the events reports but otherwise would not cause issues. This API does not support a GET.

Reply:
The "probably" part is a bit scary to me. If there is no other way to check on the server and see that you connected successfully please mention this in the detailed description of the integration so that users will know to look out for that.